src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorDataMgr.java

```diff
@@ -0,0 +1,506 @@
+    // Split the tag name into its component parts
+    List tagNameComponents = StringUtils.split(tagName, SPLIT_TAG_NAME_DELIMITER);
+
+    // Build the split tag name array
+    String[] splitTagName = new String[SPLIT_TAG_NAME_ARRAY_SIZE];
+    if (tagNameComponents.size() == 1) {
+      splitTagName[0] = null;
+      splitTagName[1] = (String) tagNameComponents.get(0);
+      splitTagName[2] = DEFAULT_SERIES_VALUE;
+    } else if (tagNameComponents.size() == 2) {
+      splitTagName[0] = null;
+      splitTagName[1] = (String) tagNameComponents.get(0);
+      splitTagName[2] = (String) tagNameComponents.get(1);
+    } else if (tagNameComponents.size() == 3) {
+      splitTagName[0] = (String) tagNameComponents.get(0);
+      splitTagName[1] = (String) tagNameComponents.get(1);
+      splitTagName[2] = (String) tagNameComponents.get(2);
+    } else if (tagNameComponents.size() > 3) {
+      splitTagName[0] = (String) tagNameComponents.get(tagNameComponents.size() - 3);
+      splitTagName[1] = (String) tagNameComponents.get(tagNameComponents.size() - 2);
+      splitTagName[2] = (String) tagNameComponents.get(tagNameComponents.size() - 1);
+    } else {
+      splitTagName[0] = null;
+      splitTagName[1] = tagName;
+      splitTagName[2] = DEFAULT_SERIES_VALUE;
+    }
+    return splitTagName;
+  }
+
+  /**
+   * Checks for historical data in the queue and sends any data points to Cumulocity.
+   *
+   * @param mqttMgr the MQTT manager to send data points on
+   */
+  public static void checkForHistoricalDataAndSend(CConnectorMqttMgr mqttMgr) {
+    // Grab latest data and send via MQTT
+    long currentTimestampMillis;
+
+    // Store current timestamp
+    currentTimestampMillis = System.currentTimeMillis();
+
+    // Update available memory variable
+    /** Current available memory in bytes */
+    long availableMemoryBytes = Runtime.getRuntime().freeMemory();
+
+    // Get queue data poll interval (millis) from config
+    long queueDataPollIntervalMillis = QUEUE_DATA_POLL_INTERVAL_MILLIS_DEFAULT;
+    try {
+      queueDataPollIntervalMillis =
+          CConnectorMain.getConnectorConfig().getQueueDataPollIntervalMillis();
+    } catch (Exception e) {
+      Logger.LOG_CRITICAL(
+          "Error getting historical data queue poll interval from configuration file. Using"
+              + " default: "
+              + queueDataPollIntervalMillis);
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    // Refresh data if within time window
+    if ((currentTimestampMillis - lastUpdateTimestampMillis) >= queueDataPollIntervalMillis) {
+      // Check if memory is within permissible range to poll data queue
+      if (availableMemoryBytes < QUEUE_DATA_POLL_MIN_MEMORY_BYTES) {
+        // Show low memory warning
+        Logger.LOG_WARN("Low memory on device, " + (availableMemoryBytes / 1000) + " MB left!");
+
+        // If low memory flag not set, set it and request garbage collection
+        if (!isMemoryCurrentlyLow) {
```

Comment:
Yes, this matches the behavior of other connectors. We check to ensure that System.gc() is only called once.

Reply:
I'm not certain but I remember discussions where we said this method call not actually running the garbage collection and only sets a flag so that garbage collection will run at the next possible time.